src/tools/miri/src/bin/miri.rs

```diff
@@ -143,6 +145,13 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
                 Arc::make_mut(&mut crate_source).rlib = Some((PathBuf::new(), PathKind::All));
                 crate_source
             };
+
+            assert!(std::ptr::fn_addr_eq(providers.layout_of, rustc_ty_utils::layout_of as
+                for<'tcx> fn(TyCtxt<'tcx>, PseudoCanonicalInput<'tcx, Ty<'tcx>>) -> Result<TyAndLayout<'tcx>, &'tcx LayoutError<'tcx>>));
+            providers.layout_of = |tcx, query| {
+                let _span = tracing::info_span!("tcx.layout_of", "query = {:?}", query.value.kind()).entered();
+                rustc_ty_utils::layout_of(tcx, query)
+            };
```

Comment:
What a hack.^^

But also, this will only measure the time inside the query, not the time spent in the query machinery, so I think this is the wrong place to add the tracing span.

Ultimately we want to catch all the `self.layout_of` calls in `InterpCx`. Maybe just add a new inherent method, which should take priority over the trait method, and do the tracing there?

Reply:
> Ultimately we want to catch all the self.layout_of calls in InterpCx. Maybe just add a new inherent method, which should take priority over the trait method, and do the tracing there?

Oh ok, thanks for clarifying. I pushed a commit that does that, and added a comment on the inherent method hoping it won't cause confusion. Here is a trace collected with both the query override and the inherent method: [trace-1751014622990785.json](https://github.com/user-attachments/files/20943205/trace-1751014622990785.json)

Should I keep the query override? I think it may still provide useful info, as by looking at the trace it seems like the `layout_of` query is sometimes called outside of `InterpCx::layout_of`.

Reply:
I'd prefer not to have the query override -- but it may be good to figure out where those "outside" calls are coming from. They may be coming from other parts of the compiler, not from Miri itself.

Reply:
Ok, I removed that commit and added a todo on my side to figure out where the "outside" calls come from